lib/spack/spack/__init__.py

```diff
@@ -214,21 +214,22 @@
 
 
 # Set up the user's editor
+_default_editors = ['vim', 'vi', 'emacs', 'nano']
+
 # $EDITOR environment variable has the highest precedence
 editor = os.environ.get('EDITOR')
 
 # if editor is not set, use some sensible defaults
 if editor is not None:
     editor = Executable(editor)
 else:
-    editor = which('vim', 'vi', 'emacs', 'nano')
+    editor = which()
 
+# Editor doesn't exist, but this will only fail when someone tries to
+# invoke it.  Spack shouldn't raise an error if the editor is not found
+# here, because it will cause non-interactive Spack jobs to fail.
 if not editor:
-    default = default_editors[0]
-    msg  = 'Default text editor, {0}, not found.\n'.format(default)
-    msg += 'Please set the EDITOR environment variable to your preferred '
-    msg += 'text editor, or install {0}.'.format(default)
-    raise EnvironmentError(msg)
+    editor = Executable(_default_editors[0])
```

Comment:
I think we're circling back around to the point where #4230 is practically reverted. At this point, `vim` is used instead of `vi`, but there is no backup plan. Some older systems might not have `vim` installed. Even Fedora only comes with a `vi` executable, which is actually `vim` in disguise.

Why not just remove this error message block and use `which` like in your previous PR?